src/puppetlabs/trapperkeeper/services/metrics/metrics_core.clj

```diff
@@ -34,24 +34,27 @@
 (def MetricsServiceContext
   {:registries (schema/atom {schema/Any RegistryContext})})
 
+(def Keyword-or-Str (schema/if keyword? schema/Keyword schema/Str))
+
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Private
 
 (schema/defn jmx-reporter :- JmxReporter
   [registry :- MetricRegistry
-   domain :- (schema/maybe schema/Str)]
+   domain :- (schema/maybe schema/Keyword)]
   (let [b (JmxReporter/forRegistry registry)]
-    (when-let [^String d domain]
-      (.inDomain b d))
+    (when domain
+      (.inDomain b (name domain)))
     (.build b)))
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
 (schema/defn initialize :- RegistryContext
   [config :- MetricsConfig
-   domain :- (schema/maybe schema/Str)]
-  (let [jmx-config (get-in config [:reporters :jmx])
+   domain :- (schema/maybe Keyword-or-Str)]
```

Comment:
It's a public function so someone could be using it, so despite it not being part of the primary API that is the TK metrics service's protocol, I think it would still be a breaking API change.